### DISCUSSION TITLE
error in pullSnpGeno() function

### DISCUSSION THREAD

**Discussion by Anilkumargpb** [gaynorr/AlphaSimR#185] [Q&A]
HI everyone,

I was  generating the founder population, including marker data, marker effects, and map information, with following codes. after a round of random crossing, the snp information of simulated population is showing zeros for all markers upon pulling snp genotypes. Here are the codes used. 
founderPop=importInbredGeno(geno = snp, genMap = GenMap, ped = NULL)
SP = SimParam$new(founderPop)
SP$importTrait(markerNames = qtlEffects$marker,
               addEff = qtlEffects$aditiveEffect,
               name = "Your_Trait")
SP$addSnpChip(5)
Parents = newPop(founderPop)
F1 = randCross(Parents, 200)
HDRW = makeDH(F1, 100)
pullSnpGeno(HDRW)

Let me know where I am going wrong. 

Thanks

**Comment by gregorgorjanc** [comment]
@Anilkumargpb `makeDH()` is creating doubled-haploid (DH) (=fully homozygous) individuals, so genotypes of all 0s makes sense, but seeing just 0s is troubling - we would expect homozygous 0 and 2 - at least in some proportion. Maybe you are only looking at some individuals and some markers? Or is your `snp` genotype matrix really sufficiently diverse?

What do these commands report:
  * `table(c(pullSnpGeno(HDRW)))` # trying to see distribution of 0s and 2s
  * `colMeans(pullSnpGeno(HDRW)) / 2 ` # trying to see allele freqs at markers

**Reply by Anilkumargpb** [reply]
HI Gregor 

Thanks for the detailed information. I found my mistake, it was due to a set of markers that are not polymorphic. 

Thanks

**Reply by Anilkumargpb** [reply]
Hi 

I have a question on the same topic, when I am using _pullSnpGeno(HDRW)_  its only giving information on a set of SNPs. How do I pull information of all markers?
Please let me know.

Thanks

**Reply by gaynorr** [reply]
The function `pullSegSiteGeno` will give you the genotype information for all the markers.

The `pullSnpGeno` function is designed to extract a subset of markers that have been assigned to a SNP chip. The function `pullMarkerGeno` will give you genotype information for specific markers you request.

**Reply by Anilkumargpb** [reply]
Thank you so much Chris
